src/NServiceBus.Gateway/Receiving/SingleCallChannelReceiver.cs

```diff
@@ -205,17 +208,17 @@ static TimeSpan GetTimeToBeReceived(IDictionary<string, string> headers)
             return result;
         }
 
-        async Task HandleDatabusProperty(CallInfo callInfo)
+        async Task HandleDatabusProperty(CallInfo callInfo, CancellationToken cancellationToken = default)
         {
             if (databus == null)
             {
                 throw new InvalidOperationException("Databus transmission received without a configured databus");
             }
 
-            var newDatabusKey = await databus.Put(callInfo.Data, callInfo.TimeToBeReceived).ConfigureAwait(false);
+            var newDatabusKey = await databus.Put(callInfo.Data, callInfo.TimeToBeReceived, cancellationToken).ConfigureAwait(false);
             if (callInfo.Md5 != null)
             {
-                using (var databusStream = await databus.Get(newDatabusKey).ConfigureAwait(false))
+                using (var databusStream = await databus.Get(newDatabusKey, cancellationToken).ConfigureAwait(false))
```

Comment:
Should the token be passed to the Hasher as well?

Reply:
Missed that. Added.